pages/data.py

```diff
@@ -94,63 +101,73 @@ def update_store_trajectories(start_date: str, end_date: str, tz: str, excluded_
-        return html.Div([html.P("Data structure error.")]), loaded_uuids_store, True
+        # Retrieve already loaded data from the store
+        loaded_data = loaded_uuids_store.get('data', [])
+        total_loaded = len(loaded_data)
 
-    # Retrieve already loaded data from the store
-    loaded_data = loaded_uuids_store.get('data', [])
-    total_loaded = len(loaded_data)
+        # Handle lazy loading
+        if not loaded_uuids_store.get('loaded', False):
+            total_to_load = total_loaded + initial_batch_size
+            total_to_load = min(total_to_load, len(uuids_list))  # Avoid loading more than available
 
-    # Handle the UUIDs tab with lazy loading
-    if tab == 'tab-uuids-datatable' and not loaded_uuids_store.get('loaded', False):
-        total_to_load = total_loaded + initial_batch_size
-        total_to_load = min(total_to_load, len(uuids_list))  # Avoid loading more than available
+            logging.debug(f"Loading next batch of UUIDs: {total_loaded} to {total_to_load}")
 
-        logging.debug(f"Loading next batch of UUIDs: {total_loaded} to {total_to_load}")
-        
-        # Slice the list of UUIDs from the dict
-        new_data = uuids_list[total_loaded:total_to_load]
+            # Slice the list of UUIDs from the dict
+            new_data = uuids_list[total_loaded:total_to_load]
 
-        if new_data:
-            # Process and append the new data to the loaded store
-            processed_data = db_utils.add_user_stats(new_data, initial_batch_size)
-            loaded_data.extend(processed_data)
+            if new_data:
+                # Process and append the new data to the loaded store
+                processed_data = db_utils.add_user_stats(new_data, initial_batch_size)
+                loaded_data.extend(processed_data)
 
-            # Update the store with the new data
-            loaded_uuids_store['data'] = loaded_data
-            loaded_uuids_store['loaded'] = len(loaded_data) >= len(uuids_list)  # Mark all data as loaded if done
+                # Update the store with the new data
+                loaded_uuids_store['data'] = loaded_data
+                loaded_uuids_store['loaded'] = len(loaded_data) >= len(uuids_list)  # Mark all data as loaded if done
 
-            logging.debug(f"New batch loaded. Total loaded: {len(loaded_data)}")
+                logging.debug(f"New batch loaded. Total loaded: {len(loaded_data)}")
 
         # Prepare the data to be displayed
         columns = perm_utils.get_uuids_columns()  # Get the relevant columns
         df = pd.DataFrame(loaded_data)
 
         if df.empty or not perm_utils.has_permission('data_uuids'):
             logging.debug("No data or permission issues.")
-            return html.Div([html.P("No data available or you don't have permission.")]), loaded_uuids_store, False
+            return html.Div([html.P("No data available or you don't have permission.")]), loaded_uuids_store, True
```

Comment:
why did this need to be changed? what is the second return parameter?

Reply:
@shankari, @JGreenlee improved the code by separating the callbacks for the trajectories switch from the main. The second return is just to store the existing uuids so that it doesn't have to be requeried; probably could go on a separate callback as well